node/pkg/guardiansigner/filesigner.go

```diff
@@ -0,0 +1,99 @@
+package guardiansigner
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"google.golang.org/protobuf/proto"
+
+	nodev1 "github.com/certusone/wormhole/node/pkg/proto/node/v1"
+	"golang.org/x/crypto/openpgp/armor" // nolint
+)
+
+type FileSigner struct {
+	keyPath    string
+	privateKey *ecdsa.PrivateKey
+}
+
+const (
+	GuardianKeyArmoredBlock = "WORMHOLE GUARDIAN PRIVATE KEY"
+)
+
+func NewFileSigner(unsafeDevMode bool, signerKeyPath string) (*FileSigner, error) {
+	fileSigner := &FileSigner{
+		keyPath: signerKeyPath,
+	}
+
+	f, err := os.Open(signerKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %w", err)
+	}
+
+	p, err := armor.Decode(f)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read armored file: %w", err)
+	}
+
+	if p.Type != GuardianKeyArmoredBlock {
+		return nil, fmt.Errorf("invalid block type: %s", p.Type)
+	}
+
+	b, err := io.ReadAll(p.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %w", err)
+	}
+
+	var m nodev1.GuardianKey
+	err = proto.Unmarshal(b, &m)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deserialize protobuf: %w", err)
+	}
+
+	if !unsafeDevMode && m.UnsafeDeterministicKey {
+		return nil, errors.New("refusing to use deterministic key in production")
+	}
+
+	gk, err := ethcrypto.ToECDSA(m.Data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to deserialize raw key data: %w", err)
+	}
+
+	fileSigner.privateKey = gk
+	return fileSigner, nil
+}
+
+func (fs *FileSigner) Sign(hash []byte) ([]byte, error) {
+
+	// Sign the hash
+	sig, err := crypto.Sign(hash, fs.privateKey)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign wormchain address: %w", err)
```

Comment:
It's kinda nice IMO to use this strategy or else use [errors.Join](https://pkg.go.dev/errors#Join) to provide some context on where the errors are coming from. Otherwise, in Go, it's easy for an error to pop-up but have no idea where it was called because errors don't come with line numbers or stack traces

Reply:
Yeah I like logging like this in general, but from what I've seen the caller of `Sign` gets the error and then wraps it in another string, both of which seem to be the same error string, so it ends up nested without any extra useful information

Reply:
For clarity I didn't like the fact that in other places we also formatted the error with string "failed to sign wormchain address", so we'd end up with an error logged that looked like:

```
failed to sign wormchain address: failed to sign wormchain address: [ERROR]
```

Anyhow we chatted in a call and the decision was to tweak the error here to something like "failed to sign hash"

Reply:
I'm going to leave this as is for now, except for updating the error message in `FileSigner.Sign()`, because I overlooked that. In some instances the guardian panics if the signing failed, and other instances it wraps the error and continues.